### DISCUSSION TITLE
Cannot convert the "SPO Destination Site Collection Url" value of type "System.String" to type "PnP.PowerShell.Commands.Base.PnPConnection".

### DISCUSSION THREAD

**Discussion by cdriskell** [pnp/powershell#2943] [Q&A]
I am reading the source url, destination url  and publishing page from a csv file.connections are verified and established successfully.  When I get to the convertTo-PnPPage command, I generate the following error from the destination SPO Url.

Cannot bind parameter 'TargetConnection'. Cannot convert the "" value of type "System.String" to type "PnP.PowerShell.Commands.Base.PnPConnection".

**Comment by AndersRask** [comment]
You need to supply a connection, not a URL, for the TargetConnection parameter.

```
-TargetConnection <PnPConnection>
        Optional connection to be used by the cmdlet. Retrieve the value for this parameter by either specifying
        -ReturnConnection on Connect-PnPOnline or by executing Get-PnPConnection.
```
